netsurf: broken functionality

## Issue description

Get netsurf onto your machine and attempt to execute it, notice how it is broken. 

Specifically nothing happens, other than the terminal (ST in this case) displays a single `\`. Trying another terminal such as Kitty or the Linux console spews a sequence of characters onto the display. Neither behavior is correct.

### Steps to reproduce

```
$ nix-shell --pure -p netsurf.browser
$ netsurf https://www.google.com
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.16, NixOS, 19.03pre166987.bc41317e243 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03pre166987.bc41317e243"`
 - channels(sna): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

@fgaz Does it still support the framebuffer? It seems to set the GTK3 as the default.

When I try to build `netsurf.browser.override { uilib="framebuffer"; }`, I get the following error:

```
/nix/store/dx8hynidprz3kf4ngcjipnwaxp6h229f-binutils-2.40/bin/ld: build/Linux-framebuffer/frontends_framebuffer_font_internal.o:/build/netsurf-3.10/build/Linux-framebuffer/font-ns-sans.h:29: multiple definition of `fb_regular_sections'; build/Linux-framebuffer/build_Linux-framebuffer_font-ns-sans.o:/build/netsurf-3.10/build/Linux-framebuffer/font-ns-sans.c:1348: first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:715: nsfb] Error 1
```

--- a/pkgs/applications/networking/browsers/netsurf/browser.nix
+++ b/pkgs/applications/networking/browsers/netsurf/browser.nix
@@ -55,6 +55,17 @@ stdenv.mkDerivation rec {
   ++ optional (uilib == "gtk3") gtk3
   ;
 
+  # Since at least 2018 AD, GCC and other compilers run in `-fno-common` mode as
+  # default, in order to comply with C standards and also get rid of some bad
+  # quality code. Because of this, many codebases that weren't updated need to
+  # be patched -- or the `-fcommon` flag should be explicitly passed to the
+  # compiler
+
+  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85678
+  # https://github.com/NixOS/nixpkgs/issues/54506
+
+  env.NIX_CFLAGS_COMPILE = "-fcommon";
+
   preConfigure = ''
     cat <<EOF > Makefile.conf
     override NETSURF_GTK_RES_PATH  := $out/share/

